neqo-crypto/src/agent.rs

```diff
@@ -50,6 +50,110 @@ use crate::{
     time::{Time, TimeHolder},
 };
 
+/// Private trait for Certificate Compression implementation
+/// # Safety
+///
+/// Use `SafeCertCompression` to implement an encoder/decoder instead.
+unsafe trait UnsafeCertCompression {
+    unsafe extern "C" fn decode_callback(
+        input: *const ssl::SECItem,
+        output: *mut ::std::os::raw::c_uchar,
+        output_len: usize,
+        used_len: *mut usize,
+    ) -> ssl::SECStatus;
+
+    unsafe extern "C" fn encode_callback(
+        input: *const ssl::SECItem,
+        output: *mut ssl::SECItem,
+    ) -> ssl::SECStatus;
+}
+
+/// The trait is used to represent a certificate compression data structure
+/// Used in order to enable Certificate Compression extension during TLS connection
+pub trait SafeCertificateCompression {
```

Comment:
The public name for this doesn't need "safe" in it.

```suggestion
pub trait CertificateCompression {
```

Reply:
Agree, changed.